transport/wsn/src/test/java/org/kaazing/gateway/transport/wsn/auth/EchoServiceAuthorizationIT2.java

```diff
@@ -0,0 +1,117 @@
+            KeyStore keyStore = null;
+            try {
+                FileInputStream fileInStr = new FileInputStream(System.getProperty("user.dir")
+                        + "/target/truststore/keystore.db");
+                keyStore = KeyStore.getInstance("JCEKS");
+                keyStore.load(fileInStr, "ab987c".toCharArray());
+            }
+            catch (Exception e) {
+                e.printStackTrace();
+            }
+
+            // @formatter:off
+            GatewayConfiguration configuration =
+                    new GatewayConfigurationBuilder()
+                        .property(HTTP_REALM_ACCEPT_OPTION.getPropertyName(), "true")
+                        .webRootDirectory(new File("src/test/webapp"))
+                        .property(Gateway.GATEWAY_CONFIG_DIRECTORY_PROPERTY,"src/test/resources/gateway/conf")
+                        .service()
+                            .accept(ACCEPT)
+                            .type("echo")
+                            .crossOrigin()
+                                .allowOrigin("*")
+                            .done()
+                            .authorization()
+                                .requireRole("AUTHORIZED")
+                            .done()
+                            .acceptOption("http.realm", "firstFactorLoginModule secondFactorLoginModule")
+                        .done()
+                        .security()
+                            .keyStore(keyStore)
+                            .realm()
+                                .name("firstFactorLoginModule")
+                                .description("username-password")
+                                .httpChallengeScheme("Basic")
+                                .authorizationMode("challenge")
+                                .loginModule()
+                                    .type("file")
+                                    .success("required")
+                                    .option("file", "jaas-config.xml")
+                                .done()
+                             .done()
+                             .realm()
+                                 .name("secondFactorLoginModule")
+                                 .description("pin")
+                                 .httpChallengeScheme("Basic")
+                                 .authorizationMode("challenge")
+                                     .loginModule()
+                                     .type("file")
+                                     .success("required")
+                                     .option("file", "pin-config.xml")
+                                 .done()
+                             .done()
+                        .done()
+                    .done();
+            // @formatter:on
+            init(configuration);
+        }
+    };
+
+    @Rule
+    public TestRule chain = createRuleChain(gateway, robot);
+
+    @Specification("basic.2factor.authorized.access.with.valid.credentials")
```

Comment:
Do you want to add a test case for the following scenario :
first realm valid credentials, second realm *invalid* credentials